pinot-common/src/test/java/com/linkedin/pinot/common/utils/TarGzCompressionUtilsTest.java

```diff
@@ -122,4 +128,61 @@ public void testEmptyDirectory()
     Assert.assertEquals(segmentFiles.length, 0);
 
   }
+
+  @Test
+  public void testBadFilePath() throws Exception {
+    File metaFile = new File(segmentDir, "metadata.properties");
+    metaFile.createNewFile();
+
+    // We guarantee metaFile isn't a directory.
+    Assert.assertFalse(metaFile.isDirectory());
+
+    File tarGzPath = new File(tarDir, SEGMENT_NAME + ".tar.gz");
+    mockIncorrectData(metaFile, tarGzPath);
+
+    try {
+      TarGzCompressionUtils.unTar(tarGzPath, untarDir);
+      Assert.fail("Did not get exception!!");
+    } catch (IOException e) {
+      Assert.assertTrue(e.getMessage().contains(TarGzCompressionUtils.EXTRACT_FILE_OUTSIDE_OF_TARGET_DIR));
+    }
+  }
+
+  private void mockIncorrectData(File nonDirFile, File tarGzPath) throws IOException {
+    FileOutputStream fOut = null;
+    BufferedOutputStream bOut = null;
+    GzipCompressorOutputStream gzOut = null;
+    TarArchiveOutputStream tOut = null;
+    try {
+      fOut = new FileOutputStream(new File(tarGzPath.getPath()));
+      bOut = new BufferedOutputStream(fOut);
+      gzOut = new GzipCompressorOutputStream(bOut);
+      tOut = new TarArchiveOutputStream(gzOut);
+      tOut.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
+
+      // Mock the file that doesn't use the correct file name.
+      String badEntryName =  "../foo/bar";
+      TarArchiveEntry tarEntry = new TarArchiveEntry(nonDirFile, badEntryName);
+      tOut.putArchiveEntry(tarEntry);
+      IOUtils.copy(new FileInputStream(nonDirFile), tOut);
+      tOut.closeArchiveEntry();
+
+    }  catch (IOException e) {
+      Assert.fail("Unexpected Exception!!");
+    } finally {
+      if (tOut != null) {
```

Comment:
use the try with resources to eliminate all the null checks and boiler plate code. https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html